### DISCUSSION TITLE
failed slot-hook and install-same=false

### DISCUSSION THREAD

**Discussion by mschnelte** [rauc/rauc#1553] [Q&A]
Hi, 

(this has been tested with rauc 1.12 - i did check release notes for 1.13 and did not find anything on that topic - apologies if this is already addressed in a more recent version)

I do have a system.conf in which i have install-same=false like so:

```
[slot.boot.1]
device=/dev/mmcblk3p2
type=ext4
parent=rootfs.1
install-same=false
```
and in the manifest i have

```
[image.boot]
filename=boot.ext4
adaptive=block-hash-index
hooks=post-install
```

If now the post-install hook fails the rauc install fails - and the slot is marked as failed in the central.raucs file.
But if I simply run rauc install a second time the slot is skipped because the image is the same, the slot hook is not executed again and the rauc installation returns success. 
Is this the wanted behavior? I would have expected that the post-install hook is executed again as the previous run failed. 
How can I make sure that the rauc installation succeeds only if all post-install hooks have finished sucessfully - also on retry?

And a second question in the same area: Is there any way I can run a post install hook (i.e. a script in the bundle) after all slots have been updated? Adding a hook to the last slot would not work if the slot content did not change and install-same=false is configured.

Thanks a lot for any help.

Matthias

**Comment by jluebbe** [comment]
> Is this the wanted behavior? I would have expected that the post-install hook is executed again as the previous run failed.
How can I make sure that the rauc installation succeeds only if all post-install hooks have finished sucessfully - also on retry?

I think this is a bug, if the post-install fails, the slot should not be considered installed correctly and it should **not** be skipped when installing again.

>  Is there any way I can run a post install hook (i.e. a script in the bundle) after all slots have been updated? Adding a hook to the last slot would not work if the slot content did not change and install-same=false is configured.

Currently not. But if the actions that you want to perform are not related to the bundle contents, a [post-install handler](https://rauc.readthedocs.io/en/latest/using.html#system-based-customization-handlers) could be the right approach.

On the other hand, support for a (non-slot) [post-install hook](https://rauc.readthedocs.io/en/latest/using.html#install-hooks) at the bundle level should be easy to add. We'd be happy to take a PR for that (or [implement it via commercial support](https://rauc.io/pages/support.html)).

**Comment by mschnelte** [comment]
Thanks for the fast reply.

I created an issue https://github.com/rauc/rauc/issues/1554
